warehouse/oidc/views.py

```diff
@@ -178,14 +213,30 @@ def _invalid(errors):
                     "code": "invalid-publisher",
                     "description": f"valid token, but no corresponding publisher ({e})",
                 }
-            ]
+            ],
+            request=request,
         )
 
+    if not isinstance(publisher, OIDCPublisher):
+        # This should be impossible, but we have to perform this type check to
+        # appease mypy otherwise we get type errors in the code after this
+        # point.
+        return _invalid(
+            errors=[
+                {
+                    "code": "invalid-publisher",
+                    "description": "valid token, but no corresponding publisher",
+                }
+            ],
+            request=request,
+        )
```

Comment:
This seems weird, does typing the `find_publisher` call fix this? Can we `assert isinstance(publisher, OIDCPublisher)` in the `try`/`except` instead?

Reply:
Yeah, this was icky.  Good idea to use `assert`.